wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```diff
@@ -362,7 +362,15 @@ public function updatedObject(array $data, IStorableObject $object, $loadValues
 
             if (\is_array($value)) {
                 $value = \array_map(function ($v) use ($method) {
-                    if (!\is_string($v) || !\file_exists($v)) {
+                    if ($v instanceof UploadFile) {
+                        if (!\file_exists($v->getLocation())) {
+                            throw new \InvalidArgumentException(
+                                "The {$method} must return an array of strings or object of class '" . UploadFile::class . "' with the valid file locations for field '{$this->getId()}'."
+                            );
+                        }
+
+                        return $v;
+                    } elseif (!\is_string($v) || !\file_exists($v)) {
                         throw new \InvalidArgumentException(
                             "The {$method} must return an array of strings with the file locations for field '{$this->getId()}'."
```

Comment:
You need to update this error message. Maybe it would be easier to just `if ($v instanceof Uploadfile) $v = $v->getLocation();`?

Reply:
This would make it impossible to use the image link-feature of `UploadFile` and setting a filename differing from the used path (e.g. attachment's original filename instead of id and hash).